### PR TITLE
Set `timeout` to `None` to avoid timeout errors

### LINKED ISSUE #187: when uploading many files: httpcore.ReadTimeout: timed out
I'm working on upgrading our "sample data" app to pyDataverse 0.3.2:

- https://github.com/IQSS/dataverse-sample-data/pull/44

@JR-1991 @jp-tosca and I have been discussing some challenges at https://dataverse.zulipchat.com/#narrow/stream/377090-python/topic/upgrading.20.22sample.20data.22.20from.20pyDataverse.200.2E2.2E1.20to.200.2E3.2E2

I'm hitting some kind of timeout when running `api.upload_datafile(dataset_pid, df_filename, df.json())`, especially when uploading [a dozen or so files](https://github.com/IQSS/dataverse-sample-data/tree/8bfeea47694e3709df49a6c193c584ba1cbe9086/data/dataverses/manchester/datasets/test-dataset/files) to a dataset.

"The default behavior is to raise a TimeoutException after 5 seconds of network inactivity" according to https://www.python-httpx.org/advanced/timeouts/

@JR-1991 suggested trying https://github.com/gdcc/pyDataverse/commit/a524c1c91732a159f580750a7af30dab5eed04a0 which adds `timeout=None` and it seems to be helping. I can create all the sample datasets and upload files, as before.

### PR DESCRIPTION
**Pull request by JR-1991**
**Overview**

In issue #187, it was noted that the current implementation using `httpx` throws an exception when a timeout exceeds 5 seconds. This can be problematic when uploading bulk data to Dataverse, as server-side processing times can cause delays and trigger the exception. This pull request sets the `timeout` parameter to `None`, which is equivalent to an unlimited timeout. We may need to consider whether this approach is reasonable or if a specific time limit would be more appropriate.

**TLDR**

* Set `timeout` to `None` for no timeout

**Tasks**

- [x] Decide if unlimited or a certain time is appropriate
- [x] Add test to upload bulk data to verify functionality

Closes #187

### CODE REVIEW

**Review by pdurbin — APPROVED** [review]
I know this a draft PR but I'm already approving it. It fixes my problem. 😄

**Comment by JR-1991** [comment]
@pdurbin thanks! Will deliver the tests soon and then we can merge it
